operator/v1/types_ingress.go

```diff
@@ -1382,6 +1382,101 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported to be set via route annotation haproxy.router.openshift.io/hsts_header and not via this API.
+	// Setting httpCaptureHeaders for the headers set via this API will not work.
+	// If spec.clientTLS, spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId, spec.httpHeaders.headerNameCaseAdjustments is set
+	// and if you set the same header via this API then that header will get replaced with the value specified via this API.
+	// If cache-control is set using this API then the existing value of cache-control will get replaced.
+	// +optional
+	Actions IngressControllerHTTPHeaderActions `json:"actions,omitempty"`
+}
+
+// IngressControllerHTTPHeaderActions defines configuration for actions on HTTP request and response headers.
+type IngressControllerHTTPHeaderActions struct {
+	// response is a list of HTTP response headers to set or delete.
+	// The actions either Set or Delete will be performed on the headers in sequence as defined in the list of response headers.
+	// A maximum of 64 response header actions may be configured.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	// +kubebuilder:validation:MaxItems=64
+	// +kubebuilder:validation:XValidation:rule="self.all(key, !has(key.action.set) || !key.action.set.value.contains('req.hdr'))",message="Response Header value can't have req.hdr converter."
```

Comment:
A small wording update
```suggestion
	// +kubebuilder:validation:XValidation:rule="self.all(key, !has(key.action.set) || !key.action.set.value.contains('req.hdr'))",message="Response header values may not use the req.hdr converter."
```